src/crawlee/browsers/_playwright_browser_controller.py

```diff
@@ -116,13 +116,11 @@ async def new_page(
 
     @override
     async def close(self, *, force: bool = False) -> None:
-        if force:
-            for page in self._pages:
-                await page.close()
-
-        if self.pages_count > 0:
+        if self.pages_count > 0 and not force:
             raise ValueError('Cannot close the browser while there are open pages.')
 
+        if self._browser_context:
+            await self._browser_context.close()
```

Comment:
Done as recommended in PW documentation.

[browser.new_context]( https://playwright.dev/python/docs/api/class-browser#browser-new-context)
"If directly using this method to create [BrowserContext](https://playwright.dev/python/docs/api/class-browsercontext)s, it is best practice to explicitly close the returned context via [browser_context.close()](https://playwright.dev/python/docs/api/class-browsercontext#browser-context-close) when your code is done with the [BrowserContext](https://playwright.dev/python/docs/api/class-browsercontext), and before calling [browser.close()](https://playwright.dev/python/docs/api/class-browser#browser-close). This will ensure the context is closed gracefully and any artifacts—like HARs and videos—are fully flushed and saved."

Also mentioned in [browser.close](https://playwright.dev/python/docs/api/class-browser#browser-close)
"This is similar to force quitting the browser. Therefore, you should call [browser_context.close()](https://playwright.dev/python/docs/api/class-browsercontext#browser-context-close) on any [BrowserContext](https://playwright.dev/python/docs/api/class-browsercontext)'s you explicitly created earlier with [browser.new_context()](https://playwright.dev/python/docs/api/class-browser#browser-new-context) before calling [browser.close()](https://playwright.dev/python/docs/api/class-browser#browser-close)."

Reply:
Thanks